Libplanet.Tests/Blocks/BlockHeaderTest.cs

```diff
@@ -12,13 +14,124 @@ public class BlockHeaderTest : IClassFixture<BlockFixture>
+                difficulty: _fx.Next.Difficulty,
+                totalDifficulty: _fx.Next.TotalDifficulty,
+                nonce: _fx.Next.Nonce.ByteArray,
+                miner: ImmutableArray<byte>.Empty,
+                hash: _fx.Next.Hash.ByteArray,
+                txHash: randomHash,
+                previousHash: _fx.Genesis.Hash.ByteArray,
+                timestamp: _fx.Next.Timestamp.ToString(
+                    BlockHeader.TimestampFormat,
+                    CultureInfo.InvariantCulture
+                ),
+                preEvaluationHash: ImmutableArray<byte>.Empty,
+                stateRootHash: randomHash2
+            );
+            expected = Bencodex.Types.Dictionary.Empty
+                .Add(BlockHeader.ProtocolVersionKey, 1)
+                .Add(BlockHeader.IndexKey, 1)
+                .Add(
+                    BlockHeader.TimestampKey,
+                    _fx.Next.Timestamp.ToString(
+                        BlockHeader.TimestampFormat,
+                        CultureInfo.InvariantCulture
+                    )
+                )
+                .Add(BlockHeader.DifficultyKey, _fx.Next.Difficulty)
+                .Add(
+                    BlockHeader.TotalDifficultyKey,
+                    (IValue)new Bencodex.Types.Integer(_fx.Next.TotalDifficulty)
+                )
+                .Add(BlockHeader.NonceKey, _fx.Next.Nonce.ToByteArray())
+                .Add(BlockHeader.HashKey, _fx.Next.Hash.ToByteArray())
+                .Add(BlockHeader.PreviousHashKey, _fx.Genesis.Hash.ToByteArray())
+                .Add(BlockHeader.TxHashKey, randomHash.ToArray())
+                .Add(BlockHeader.StateRootHashKey, randomHash2.ToArray());
+            Assert.Equal(expected, header.ToBencodex());
+            Assert.Equal(expected, new BlockHeader(expected).ToBencodex());
+        }
+
+        [Fact]
+        public void ValidateProtocolVersion()
+        {
+            var header = new BlockHeader(
+                protocolVersion: -1,
+                index: 0,
+                difficulty: _fx.Next.Difficulty,
+                totalDifficulty: _fx.Next.TotalDifficulty,
+                nonce: _fx.Next.Nonce.ByteArray,
+                miner: _fx.Next.Miner?.ByteArray ?? ImmutableArray<byte>.Empty,
+                hash: _fx.Next.Hash.ByteArray,
+                txHash: _fx.Next.TxHash?.ByteArray ?? ImmutableArray<byte>.Empty,
+                previousHash: _fx.Next.PreviousHash?.ByteArray ?? ImmutableArray<byte>.Empty,
+                timestamp: _fx.Next.Timestamp.ToString(
+                    BlockHeader.TimestampFormat,
+                    CultureInfo.InvariantCulture
+                ),
+                preEvaluationHash: TestUtils.GetRandomBytes(32).ToImmutableArray(),
+                stateRootHash: ImmutableArray<byte>.Empty
+            );
+
+            Assert.Throws<InvalidBlockProtocolVersionException>(() =>
+                header.Validate(DateTimeOffset.UtcNow)
+            );
```

Comment:
Could we have a more test case where valid block protocol version? Like https://github.com/planetarium/libplanet/pull/1147/files#diff-b70ec4dc8f1efa4e59f3584b07849b9155f1f3d448a2e026a5d80e61c3de4bf5R152

Reply:
The link seems lost.

Reply:
Like this https://github.com/planetarium/libplanet/blob/main/Libplanet.Tests/Blocks/BlockHeaderTest.cs#L38-L39

Reply:
Added:

```diff
--- a/Libplanet.Tests/Blocks/BlockHeaderTest.cs
+++ b/Libplanet.Tests/Blocks/BlockHeaderTest.cs
@@ -98,6 +98,13 @@ namespace Libplanet.Tests.Blocks
             Assert.Equal(expected, new BlockHeader(expected).ToBencodex());
         }
 
+        [Fact]
+        public void ValidateValidHeader()
+        {
+            _fx.Genesis.Header.Validate(DateTimeOffset.UtcNow);
+            _fx.Next.Header.Validate(DateTimeOffset.UtcNow);
+        }
+
         [Fact]
         public void ValidateProtocolVersion()
         {
```